test/visualization/test_plotter_mpldrawer.py

```diff
@@ -101,7 +101,11 @@ def test_unit_scale(self, args):
     def test_scale(self):
         """Test the xscale and yscale figure options."""
         plotter = MockPlotter(MplDrawer(), plotting_enabled=True)
-        plotter.set_figure_options(xscale="quadratic", yscale="log")
+        plotter.set_figure_options(
```

Comment:
Outside the scope of this PR, but we should probably handle the case of a log scale with limit 0 gracefully.

Reply:
I decided to address this here (see [aed5d30](https://github.com/Qiskit-Extensions/qiskit-experiments/pull/1351/commits/aed5d307d1f1d501f2f90c1a0846c2c609c370a4)). I was able to find a somewhat elegant solution by moving the setting of the axis scale earlier in the process (see the commit message). The tricky part is that the code wants to know the axis limits for the part where it tries to rescale the format (like add a k and hide the 000) but if you have axes that have no data in them matplotlib returns 0,1 by default. If the code didn't do that formatting scaling, I would just change it to not set limits when no user limits were specified so matplotlib could use its default limits.